images/access-setup/content/bin/common.sh

```diff
@@ -37,8 +37,16 @@ get_context() {
     printf "[ERROR] Install jq\n" >&2
     exit 1
   fi
-  token_secret="$(KUBECONFIG="$target" kubectl get sa "$sa" -n "$namespace" -o json |
-    jq -r '.secrets[].name | select(. | test(".*token.*"))')"
+
+  token_secret=""
+  while [ -z "$token_secret" ]; do
+    token_secret="$(KUBECONFIG="$target" kubectl get sa "$sa" -n "$namespace" -o json |
+    jq -r '.secrets[]?.name | select(. | test(".*token.*"))')"
+    if [ -n "$token_secret" ]; then
```

Comment:
I believe that bit should not be necessary, since on the next while loop the end condition should be met.

Reply:
so we just use `sleep 5` to wait a while, right?

Reply:
"sleep 1" or "sleep 2" could be an enhancement but it is unrelated to Romain's point.
Romain's point was to remove:
~~~
    if [ -n "$token_secret" ]; then
      break
    fi
~~~
as with each iteration `[ -z "$token_secret" ]`is evaluated (right after the above code) and if the variable has been populated the loop is stopped

Reply:
ah, thank you for the explanation . fixed.